File: Leetcode/201. Bitwise AND of Numbers Range/bitwise_AND_of_numbers_range.py
class Solution:
    def rangeBitwiseAnd(self, left: int, right: int) -> int:
        count = 0
        while left !=right:
            left >>= 1
            right >>= 1
            count +=1
        return left << count
    
    def rangeBitwiseAnd1(self, left: int, right: int) -> int:
        ans = right
        while ans and left < right:
            ans &= left
            left+=1
        return ans

            
    
item  = Solution()
print(item.rangeBitwiseAnd(5,7))
print(item.rangeBitwiseAnd1(5,7))

File: Leetcode/201. Bitwise AND of Numbers Range/bitwise_AND_of_numbers_range.rs
struct Solution{}

impl Solution {
    pub fn range_bitwise_and(mut left: i32, mut right: i32) -> i32 {
        let mut count =0;
        while left !=right{
            left>>=1;
            right>>=1;
            count+=1
        }
        left<<count
        
    }
}
fn main() {
    let res = Solution::range_bitwise_and(5,7);
    println!("{}",res)
}
